Not specifying a calibration file with URSim throws a warning

When using the driver with a URSim instance, I get the warning

```
[ros2_control_node-1] [ERROR] [1639130909.694879192] [UR_Client_Library]: The calibration parameters of the connected robot don't match the ones from the given kinematics config file. Please be aware that this can lead to critical inaccuracies of tcp positions. Use the ur_calibration tool to extract the correct calibration from the robot and pass that into the description. See [https://github.com/UniversalRobots/Universal_Robots_ROS_Driver#extract-calibration-information] for details.
```

In the ROS1 driver we use default values that match the parameters from ursim (which is basically a description where all "correction" parameters are 0). As the [default_kinematics.yaml](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_description/config/ur5e/default_kinematics.yaml) is the same in the ROS2 repository, I assume that the intention was to also reflect this behavior in the ROS2 driver.

Tested with a UR5e ursim version 5.11.0-108249.

On a first glance, everything looks like it gets passed around correctly, so I would have to further debug this to find the issue. However, the ROS2 driver uses the deprecated method of passing in the calibration to the library's `UrDriver` constructor. We have a separate function to check the calibration checksum since a while. See the [ROS1 hw interface](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/src/hardware_interface.cpp#L296-L309) for reference.

--- a/ur_robot_driver/src/hardware_interface.cpp
+++ b/ur_robot_driver/src/hardware_interface.cpp
@@ -384,7 +384,8 @@ CallbackReturn URPositionHardwareInterface::on_activate(const rclcpp_lifecycle::
     RCLCPP_FATAL_STREAM(rclcpp::get_logger("URPositionHardwareInterface"), e.what());
     return CallbackReturn::ERROR;
   }
-
+  RCLCPP_INFO(rclcpp::get_logger("URPositionHardwareInterface"), "Calibration checksum: '%s'.",
+              calibration_checksum.c_str());
   // check calibration
   // https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/c3378599d5fa73a261328b326392e847f312ab6b/ur_robot_driver/src/hardware_interface.cpp#L296-L309
   if (ur_driver_->checkCalibration(calibration_checksum)) {

--- a/ur_robot_driver/src/hardware_interface.cpp
+++ b/ur_robot_driver/src/hardware_interface.cpp
@@ -374,8 +374,8 @@ CallbackReturn URPositionHardwareInterface::on_activate(const rclcpp_lifecycle::
     ur_driver_ = std::make_unique<urcl::UrDriver>(
         robot_ip, script_filename, output_recipe_filename, input_recipe_filename,
         std::bind(&URPositionHardwareInterface::handleRobotProgramState, this, std::placeholders::_1), headless_mode,
-        std::move(tool_comm_setup), calibration_checksum, (uint32_t)reverse_port, (uint32_t)script_sender_port,
-        servoj_gain, servoj_lookahead_time, non_blocking_read_);
+        std::move(tool_comm_setup), (uint32_t)reverse_port, (uint32_t)script_sender_port, servoj_gain,
+        servoj_lookahead_time, non_blocking_read_);
   } catch (urcl::ToolCommNotAvailable& e) {
     RCLCPP_FATAL_STREAM(rclcpp::get_logger("URPositionHardwareInterface"), "See parameter use_tool_communication");
 
@@ -383,6 +383,27 @@ CallbackReturn URPositionHardwareInterface::on_activate(const rclcpp_lifecycle::
   } catch (urcl::UrException& e) {
     RCLCPP_FATAL_STREAM(rclcpp::get_logger("URPositionHardwareInterface"), e.what());
     return CallbackReturn::ERROR;
+  }
+
+  // check calibration
+  // https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/c3378599d5fa73a261328b326392e847f312ab6b/ur_robot_driver/src/hardware_interface.cpp#L296-L309
+  if (ur_driver_->checkCalibration(calibration_checksum)) {
+    RCLCPP_INFO(rclcpp::get_logger("URPositionHardwareInterface"), "Calibration checked successfully.");
+  } else {
+    RCLCPP_ERROR_STREAM(rclcpp::get_logger("URPositionHardwareInterface"), "The calibration parameters of the "
+                                                                           "connected robot don't match the ones from "
+                                                                           "the given kinematics "
+                                                                           "config file. Please be aware that this can "
+                                                                           "lead to critical inaccuracies of tcp "
+                                                                           "positions. Use "
+                                                                           "the ur_calibration tool to extract the "
+                                                                           "correct calibration from the robot and "
+                                                                           "pass that into the "
+                                                                           "description. See "
+                                                                           "[https://github.com/UniversalRobots/"
+                                                                           "Universal_Robots_ROS_Driver#extract-"
+                                                                           "calibration-information] "
+                                                                           "for details.");
   }
 
   ur_driver_->startRTDECommunication();

